analysis/org.eclipse.tracecompass.analysis.timing.core.tests/src/org/eclipse/tracecompass/analysis/timing/core/tests/segmentstore/AbstractSegmentStoreStatisticsDataProviderTest.java

```diff
@@ -0,0 +1,233 @@
+            assertEquals("Average (index " + index + ")", expectedEntries.get(i).fAverage, entry.getMean(), 0.02);
+            assertEquals("StdDev (index " + index + ")", expectedEntries.get(i).fStdDev, entry.getStdDev(), 0.02);
+            assertEquals("Count (index " + index + ")", expectedEntries.get(i).fNbElements, entry.getNbElements());
+            assertEquals("Total (index " + index + ")", expectedEntries.get(i).fTotal, entry.getTotal(), 0.02);
+
+            assertEquals("Min start (index " + index + ")", expectedEntries.get(i).fMinStart, entry.getMinStart());
+            assertEquals("Min end (index " + index + ")", expectedEntries.get(i).fMinEnd, entry.getMinEnd());
+            assertEquals("Max start (index " + index + ")", expectedEntries.get(i).fMaxStart, entry.getMaxStart());
+            assertEquals("Max end (index " + index + ")", expectedEntries.get(i).fMaxEnd, entry.getMaxEnd());
+        }
+    }
+
+    /**
+     * Verifies the entries with user-defined aspects in the segment store
+     * statistics against the expected labels and entries.
+     *
+     * @param expectedLabels
+     *            The expected labels for each entry.
+     * @param expectedEntriesUserDefined
+     *            The expected statistics holders with user-defined fields.
+     * @param entries
+     *            The actual segment store statistics entries.
+     * @param startIndex
+     *            The start index for verification.
+     * @param nbEntries
+     *            The number of entries to verify.
+     */
+    protected static void verifyEntriesWithUserDefinedAspect(List<@NonNull List<@NonNull String>> expectedLabels,
+            @NonNull List<@NonNull StatisticsHolderUserDefined> expectedEntriesUserDefined, List<@NonNull SegmentStoreStatisticsModel> entries, int startIndex, int nbEntries) {
+        @NonNull List<@NonNull StatisticsHolder> expectedEntries = expectedEntriesUserDefined.stream().collect(Collectors.toList());
+        verifyEntries(expectedLabels, expectedEntries, entries, startIndex, nbEntries);
+        for (int i = 0; i < expectedLabels.size(); i++) {
+            String expectedUserDefinedValue = expectedEntriesUserDefined.get(i).fUserDefinedField;
+            int index = startIndex + i;
+            SegmentStoreStatisticsModel entry = entries.get(index);
+            assertTrue("User defined aspect not found (index " + index + ")", entry.getLabels().stream().anyMatch(s -> s.equals(expectedUserDefinedValue)));
+        }
+    }
+
+    /**
+     * Resets static ENTRY_ID of AbstractSegmentStoreStatisticsDataProvider, so
+     * that each test class extending
+     * AbstractSegmentStoreStatisticsDataProviderTest has known IDs, independent
+     * of the order of execution.
+     */
+    protected static void resetIds() {
+            try {
+                Field f = AbstractSegmentStoreStatisticsDataProvider.class.getDeclaredField("ENTRY_ID");
```

Comment:
> Sorry, I proposed the method as is. I should have made sure that I don't have single letter variables. Please update the variable names to something more meaningful since you the formatting needs to be updated anyways. The variables in question are `f`, `o` and `m`.